tests/ashley/api/test_api_upload_image.py

```diff
@@ -0,0 +1,203 @@
+        self.assertEqual(content, {"detail": 'Method "GET" not allowed.'})
+
+    def test_access_authenticated_api_upload_images_item(self):
+        """Authenticated users should not be allowed to retrieve image detail"""
+        user = UserFactory()
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        image = UploadImageFactory()
+        response = self.client.get(f"/api/v1.0/images/{image.id}/")
+        self.assertEqual(response.status_code, 404)
+
+    def test_access_authenticated_api_create_upload_image(self):
+        """Authenticated users should be able to create an image but request
+        needs to be completed."""
+        user = UserFactory()
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        response = self.client.post("/api/v1.0/images/")
+        # Request uncomplete so it returns a 400 but not a 403 anymore
+        self.assertEqual(response.status_code, 400)
+
+    def _generate_image(self, filename):
+        """Generates image of 1 px"""
+        image_file = BytesIO()
+        image = Image.new("RGBA", size=(1, 1), color=(256, 256, 256))
+        image.save(image_file, "png")
+        image_file.seek(0)
+        return ContentFile(image_file.read(), filename)
+
+    def test_api_create_upload_image(self):
+        """Verify that image gets created and uploaded an amazon"""
+        user = UserFactory()
+        forum = ForumFactory()
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        filename = f"test-{datetime.datetime.now().strftime('%f')}.png"
+        data = {
+            "file": self._generate_image(filename),
+            "forum": forum.id,
+            "poster": user.id,
+        }
+        # At first, no image exists
+        self.assertEqual(UploadImage.objects.count(), 0)
+        # Upload image
+        response = self.client.post("/api/v1.0/images/", data, format="multipart")
+        self.assertEqual(response.status_code, 201)
+
+        # Gets last id generated
+        image_db = UploadImage.objects.last()
+
+        # Image should have be added with right values
+        self.assertEqual(UploadImage.objects.count(), 1)
+        self.assertEqual(image_db.forum.id, forum.id)
+        self.assertEqual(image_db.poster.id, user.id)
+
+        # Controls image's path is respected on server
+        self.assertIn(f"image_uploads/{forum.id}/{user.id}/", str(image_db.file))
+
+        # Request library doesn't know how to handle local files
+        if settings.DEFAULT_FILE_STORAGE == "storages.backends.s3boto3.S3Boto3Storage":
+            # Controls url returned exists and contains data
+            request = urllib.request.Request(
+                f"http://{settings.AWS_STORAGE_BUCKET_NAME}.s3.amazonaws.com/{str(image_db.file)}"
+            )
+            fileHandle = urllib.request.urlopen(request)
+            data = fileHandle.read()
+            self.assertTrue(data is not None)
```

Comment:
This test only works for the S3Boto3Storage backend.
I propose the following alternative that delegates the retrieval of the file content to the backend storage.
And it checks that the file content is not empty:

```suggestion
        uploaded_file = image_db.file.open()
        self.assertGreater(uploaded_file.size, 0)
        self.assertIsNotNone(uploaded_file.read())
```

This way, it will work with all the storage backends.